webdriver-spec.html

```diff
@@ -5777,19 +5778,25 @@
  <li><p>Let <var>body</var> and <var>arguments</var>
   be <var>script arguments</var>’ data.
 
- <li><p>Let <var>result</var> be the result of calling <a>execute a function body</a>,
-  with arguments <var>body</var> and <var>arguments</var>.
+ <li><p>Let <var>promise</var> be a new Promise.
 
- <li><p>Wait for <a>execute a function body</a> to return
-  or for <a>session script timeout</a> milliseconds to expire,
-  whichever happens sooner.
+ <li><p>Run the following substeps <a>in parallel</a>:
+   <ol>
+     <li><p>Let <var>result</var> be the result of calling <a>execute a function body</a>,
+      with arguments <var>body</var> and <var>arguments</var>.
 
- <li><p>If the previous step completed due to the <a>session script timeout</a> being reached,
-  return <a>error</a> with <a>error code</a> <a>script timeout</a>.
+     <li><p>If <var>promise</var> is still pending and
+       <a>session script timeout</a> milliseconds is reached,
+      reject <var>promise</var> with <a>error</a> with
+      <a>error code</a> <a>script timeout</a>.
 
- <li><p>If <var>result</var> is an <a>error</a>, return result.
+     <li><p>If <var>result</var> is an <a>error</a>,
```

Comment:
This sounds like you're saying:

```js
new Promise((resolve, reject) => {
  let result = fn();
  if (result instanceof Error) {
    reject(result);
  }
});
```

Instead of

```js
new Promise((resolve, reject) => {
  try {
    let result = fn();
  } catch (ex) {
    reject(ex);
  }
});

```